bcda/auth/client/ssas.go

```diff
@@ -393,18 +393,20 @@ func (c *SSASClient) GetVersion() (string, error) {
 	if err != nil {
 		return "", err
 	}
+	if resp.StatusCode != 200 {
+		return "", errors.New("SSAS server failed to return version ")
+	}
 	defer resp.Body.Close()
```

Comment:
We should move this defer ahead of the resp.StatusCode check. We'll want to ensure that response body is always closed.